File: 0268-missing-number/0268-missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int sum= 0, calSum, n = nums.size();
        
        calSum = n*(n+1)/2;
        for(auto it: nums){
            sum += it;
        }
        
        return calSum - sum;
        
    }
};

File: 0268-missing-number/0268-missing-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let n = nums.length;
    const val1 = (n * (n+1))/2;
    
    const val2 = nums.reduce((sum, val) => {
        sum += val;
        return sum;
    }, 0);
    
    return val1-val2;
};